test/inductor/test_cpu_repro.py

```diff
@@ -1560,6 +1561,136 @@ def fn(x):
             metrics.reset()
             self.common(fn, (value,))
 
+    @unittest.skipIf(
+        platform.machine() != "s390x" or not codecache.valid_vec_isa_list(),
+        "Does not support vectorization or not s390x machine",
+    )
+    @patch("torch.cuda.is_available", lambda: False)
+    def test_auto_zvec_simd(self):
+        zvec = codecache.valid_vec_isa_list()[0]
+        self.assertTrue(zvec.bit_width() == 256)
+
+        with config.patch({"cpp.simdlen": 0}):
+            isa = codecache.pick_vec_isa()
+            self.assertFalse(isa)
+
+        with config.patch({"cpp.simdlen": 1}):
+            isa = codecache.pick_vec_isa()
+            self.assertFalse(isa)
+
+        with config.patch({"cpp.simdlen": 257}):
+            isa = codecache.pick_vec_isa()
+            self.assertFalse(isa)
+
+        with config.patch({"cpp.simdlen": 256}):
+            isa = codecache.pick_vec_isa()
+            self.assertTrue(isa == zvec)
+
+        pre_var = os.getenv("ATEN_CPU_CAPABILITY")
+        if pre_var:
+            os.environ.pop("ATEN_CPU_CAPABILITY")
+
+        with config.patch({"cpp.simdlen": None}):
+            isa = codecache.pick_vec_isa()
+            self.assertTrue(isa == zvec)
+
+        try:
+            with config.patch({"cpp.simdlen": None}):
+                os.environ["ATEN_CPU_CAPABILITY"] = "avx2"
+                isa = codecache.pick_vec_isa()
+                self.assertTrue(isa == zvec)
```

Comment:
why isa is zvec when cpu capability is set to either avx2 or avx512?

Reply:
avx2 is invalid for zvec machine, and it will be ignored. So the picked isa is also determined automatically. This is same as eager.